_inc/lib/class-jetpack-podcast-helper.php

```diff
@@ -10,15 +10,58 @@
  */
 class Jetpack_Podcast_Helper {
 	/**
-	 * Gets a list of tracks for the supplied RSS feed. This function is used
-	 * in both server-side block rendering and in API `WPCOM_REST_API_V2_Endpoint_Podcast_Player`.
+	 * Gets podcast data formatted to be used by the Podcast Player block in both server-side
+	 * block rendering and in API `WPCOM_REST_API_V2_Endpoint_Podcast_Player`.
 	 *
 	 * @param string $feed     The RSS feed to load and list tracks for.
 	 * @param int    $quantity Optional. The number of tracks to return.
+	 * @return array|WP_Error The player data or a error object.
+	 */
+	public static function get_player_data( $feed, $quantity = 10 ) {
+		// Load feed.
+		$rss = self::load_feed( $feed );
+		if ( is_wp_error( $rss ) ) {
+			return $rss;
+		}
+
+		// Get tracks.
+		$tracks = self::get_track_list( $rss );
+
+		// Get podcast meta.
+		$cover = $rss->get_image_url();
+		$title = $rss->get_title();
+		$link  = $rss->get_link();
+
+		return array(
+			'title'  => $title,
+			'link'   => $link,
+			'cover'  => $cover,
+			'tracks' => $tracks,
```

Comment:
We escape that data on the PHP side, but should we escape it here so we get escaped data in the player in the editor as well, when the data is queried via the endpoint?

Maybe this would also be a good place to add some default values when SimplePie returns `null` values for this info?

Reply:
on it

Reply:
I have updated the logic to have explicit null or empty strings based on the value and added text processing in order to account for both the editor (via API) and the server side render.